cpp/ql/test/query-tests/Likely Bugs/Format/NonConstantFormat/nested.cpp

```diff
@@ -18,7 +18,7 @@ extern "C" int snprintf ( char * s, int n, const char * format, ... );
 struct A {
   void do_print(const char *fmt0) {
     char buf[32];
-    snprintf(buf, 32, fmt0); // BAD [should detect at top-most call]
+    snprintf(buf, 32, fmt0); // GOOD
```

Comment:
Do you mean `BAD`? This line is still flagged by the query. We've been a bit back and forth about formats from parameters being passed to non-`vprintf` functions, and I think you've convinced me that it's `BAD`.

Overall, it looks like this file has come out of sync with its entries in `NonConstantFormat.expected`. Please go through this file (and the others) to ensure that the comment starts with `// BAD` if and only if the line is listed in the `expected` file.

Reply:
I had the opposite recollection, i.e., that we should leave things as they are (wrt handling `vprintf`-like and non-`vprintf`-like calls). Also, is it true that `// BAD` always imply an alert being signaled?  In `nested.cpp` above, I (originally) marked line 21 as `// BAD` since the alert was in a the wrong place (i.e., a false positive), but I also marked line 42 as '// BAD' since that is where the alert _should_ have gone (a false negative).  Am I getting my nomenclature mixed up?

Reply:
I will tentatively mark line 21 as `// BAD [FALSE POSITIVE]` and line 42 as `// GOOD [NOT DETECTED]`.

Reply:
> Please go through this file (and the others) to ensure that the comment starts with `// BAD` if and only if the line is listed in the `expected` file.

What I wrote there did not make sense, sorry. Let me try again.

We write `// BAD` to mean that the format call ought to be flagged by a perfect query. If it's _not_ actually flagged in the `expected` file, we add `[NOT DETECTED]`. We write `// GOOD` to mean that the format call should _not_ be flagged by a perfect query. If it's actually flagged in the `expected` file, we add `[FALSE POSITIVE]`.

Reply:
Yes, that makes more sense, thanks.  I still think we should use `\\ FLAGGED` and `\\ NOT FLAGGED`, but that's a discussion for another day.